io/disk/arcconf/arcconf_raid_oper.py

```diff
@@ -0,0 +1,232 @@
+        """
+        function which decides on RAID level
+        TODO 50, 60 and 5EE
+        """
+        self.log.info(" Raid creation with %s Drives, option %s"
+                      % (cnt, type_name1))
+        if cnt == 2:
+            raid_level = ["0", "1"]
+        elif cnt == 3:
+            raid_level = ["1E"]
+        elif cnt == 4:
+            raid_level = ["10", "5", "6"]
+
+        for raid_type in raid_level:
+            self.raid_exec(type_name1, disk_data, cnt, raid_type, pair)
+            time.sleep(10)
+
+    def raid_exec(self, name1, dsk_data, cnt, raid, pair1):
+        """
+        function to create different raid functions
+        """
+        for val1 in pair1:
+            # Format all the physical disk drives
+            self.log.info(" Formatting physical drives ==>  %s"
+                          % val1)
+            cmd = "echo y | arcconf task start %s device %s %s \
+                  INITIALIZE" % (self.crtl_no, self.channel_no, val1)
+            self.check_pass(cmd, "Failed to Format drive")
+
+        self.log.info(" Creating RAID %s with drives %s"
+                      % (raid, dsk_data))
+        cmd = "echo y | arcconf create %s LOGICALDRIVE %s MAX %s %s" \
+            % (self.crtl_no, name1, raid, dsk_data)
+        self.check_pass(cmd, "Failed to create RAID %s" % raid)
+
+        if raid == 0:
+            self.format_logical(cnt)
+        self.make_fs()
+        time.sleep(10)
+
+        self.log.info(" Deleting RAID %s with drives %s"
+                      % (raid, dsk_data))
+        cmd = "echo y | arcconf delete %s logicaldrive %s" % \
+            (self.crtl_no, self.logicaldrive)
+        self.check_pass(cmd, "Failed to delete RAID %s" % raid)
+
+    def format_logical(self, count1):
+        """
+        format logical drive
+        """
+        self.log.info("Formatting Logical drive %s, having %s drives"
+                      % (self.logicaldrive, count1))
+        cmd = "echo y | arcconf task start %s LOGICALDRIVE %s CLEAR" \
+            % (self.crtl_no, self.logicaldrive)
+        self.check_pass(cmd, "Failed to format Logical Drive")
+        cmd = "arcconf getconfig %s LD" % self.crtl_no
+        self.check_pass(cmd, "Failed to display Logical device details")
+
+    def make_fs(self):
+        """
+        creates filesystem
+        """
+        cmd = "lsscsi  | grep LogicalDrv | awk \'{print $7}\'"
+        drive = self.cmdop_list(cmd)
+
+        self.log.info("Creating file system on %s" % (drive))
+
+        cmd1 = "echo y | mkfs.ext4 %s" % drive
+        self.check_pass(cmd1, "Failed to create filesystem")
+
+        cmd2 = "mount %s /mnt" % drive
+        self.check_pass(cmd2, "Failed to mount filesystem")
+
+        cmd3 = "cd /mnt && dd if=/dev/random of=Gfile.txt bs=3M count=1"
+        self.check_pass(cmd3, "Failed to create file")
+
+        cmd4 = "cd / && umount %s" % drive
+        self.check_pass(cmd4, "Failed to unmount file")
```

Comment:
@manvanthar one help i think all four check can be inter related 

like 1- if file system not created it not able to mount 
2- if it not mounted in not able to create any file in mount point 
3- if device not mounted device not able to mount 

can it will handle in if else

```
if (file system create == PASS):
else if (mounted == PASS):
  else if (file write == PASS):
```
this is just a sample please consider